### DISCUSSION TITLE
yarn up difficulties

### DISCUSSION THREAD

**Discussion by jeffrson** [yarnpkg/berry#3414] [Q&A]
Hi,

how can I use `yarn up` to upgrade *all* packages?

AFAICT, `yarn up *` or even `yarn up \*` does not update scoped packages. Even worse, it fails if there are no packages without scope:
> Usage Error: Pattern * doesn't match any packages referenced by any workspace

On the other hand, `yarn up @*/*` fails, if there's no scoped package:
> Usage Error: Pattern @\*/\* doesn't match any packages referenced by any workspace

`yarn up @*/* *` fails as well in either case.

This makes it hard to automatically update all packages in various folders with the same script. Even if I was going to ignore the error code that is returned from the command, the error message itself cannot be suppressed, because it is written to stdout, not stderr . and stdout should be seen.

BTW, the documentation does not mention the difference between "scoped" or "unscoped".

So what's the right way to deal with it?

**Comment by arcanis** [comment]
`yarn up '**/*'`, iirc

**Reply by jeffrson** [reply]
Thanks - unfortunately this gives an "Internal Error" (which looks like a bug?):

```
$ yarn up '**/*'
Internal Error: Invalid descriptor (**/*)
    at Object.Zo (...\.yarn\releases\yarn-sources.cjs:243:11997)
    at Ru.executeUpClassic (...\.yarn\releases\yarn-sources.cjs:436:2500)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Ru.execute (...\.yarn\releases\yarn-sources.cjs:436:1197)
    at async Ru.validateAndExecute (...\.yarn\releases\yarn-sources.cjs:198:620)
    at async ss.run (...\.yarn\releases\yarn-sources.cjs:212:1846)
    at async ss.runExit (..\.yarn\releases\yarn-sources.cjs:212:2013)
    at async i (...\.yarn\releases\yarn-sources.cjs:311:12501)
    at async r (...\.yarn\releases\yarn-sources.cjs:311:10741)
```

$ yarn -v
3.1.0-rc.2.git.20210907.hash-da248e49

**Reply by merceyz** [reply]
```sh
yarn up "*" "@*/*"
```
This should update both but if you want to update all dependencies within their ranges just deleting the lockfile and running an install would do that

**Reply by jeffrson** [reply]
It only updates if there **are** both un-scoped and scoped packages, unfortunately.

I have a script that runs through several folders. Those folders contain "meta dependencies" (pull in actual packages from npm), some only scoped, others only unscoped 😭

BTW, shouldn't errors be reported via stderr?

**Reply by jeffrson** [reply]
Oh, looks like
`yarn up '**'`
does it!

Great!

So there remain the unrelated "Internal error" and usage of stderr if you have time to look at it ;-)

The actual question is answered, however. Thanks!!!

**Comment by Shayan-To** [comment]
I use `yarn up **` and it seems to work for all packages.

**Comment by sawilde** [comment]
Though not in the docs I found I needed to use quotes ie I think the leading @ confuses the parsing when dealing with packages in a named scope

```
yarn up "@aws-sdk/*"@^3.496.0
```
